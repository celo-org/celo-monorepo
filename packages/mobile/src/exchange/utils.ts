import BigNumber from 'bignumber.js'
import { CURRENCY_ENUM } from 'src/geth/consts'
import { convertToContractDecimals } from 'src/tokens/saga'
import { getContractKitAsync } from 'src/web3/contracts'

export async function celoTocUsd(celoAmount: BigNumber) {
  const celoAmountInWei = (
    await convertToContractDecimals(celoAmount, CURRENCY_ENUM.GOLD)
  ).integerValue()

  const contractKit = await getContractKitAsync()
  const exchange = await contractKit.contracts.getExchange()
  const exchangeRate = await exchange.getGoldExchangeRate(celoAmountInWei)
  const cUsdAmount = exchangeRate.multipliedBy(celoAmount)

  return cUsdAmount
}
