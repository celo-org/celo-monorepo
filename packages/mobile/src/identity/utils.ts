import BigNumber from 'bignumber.js'

export function isUserBalanceSufficient(
  userBalance: string | number | null,
  estimatedTxAmount: number | string
) {
  if (!userBalance || new BigNumber(userBalance).isLessThan(new BigNumber(estimatedTxAmount))) {
    return false
  }

  return true
}
