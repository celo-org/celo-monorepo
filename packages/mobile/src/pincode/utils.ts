import { getContractKitOutsideGenerator } from 'src/web3/contracts'
import { readPrivateKeyFromLocalDisk } from 'src/web3/privateKey'

export const PIN_LENGTH = 6

export function isPinValid(pin: string) {
  return pin.length === PIN_LENGTH
}

export function isPinCorrect(
  pin: string,
  fornoMode: boolean,
  currentAccount: string
): Promise<typeof pin> {
  return new Promise((resolve, reject) => {
    if (fornoMode) {
      readPrivateKeyFromLocalDisk(currentAccount, pin)
        .then(() => resolve(pin))
        .catch(reject)
    } else {
      getContractKitOutsideGenerator()
        .then((contractKit: any) =>
          contractKit.web3.eth.personal
            .unlockAccount(currentAccount, pin, 1)
            .then((result: boolean) => (result ? resolve(pin) : reject()))
        )
        .catch(reject)
    }
  })
}
