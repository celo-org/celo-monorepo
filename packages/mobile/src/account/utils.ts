import { ADDRESS_LENGTH } from 'src/exchange/reducer'

export const isAddressFormat = (content: string): boolean => {
  return content.startsWith('0x') && content.length === ADDRESS_LENGTH
}
