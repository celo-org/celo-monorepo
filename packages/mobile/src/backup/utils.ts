import { sampleSize } from 'lodash'
import { AsyncStorage } from 'react-native'
import { generateMnemonic, wordlists } from 'react-native-bip39'
import { getKey, setKey } from 'src/utils/keyStore'
import Logger from 'src/utils/Logger'

const TAG = 'Backup/utils'

export async function createQuizWordList(mnemonic: string, language: string | null) {
  const disallowedWordSet = new Set(mnemonic.split(' '))
  const languageWordList = getWordlist(language)
  const wordOptions: string = await generateMnemonic(1000, null, languageWordList)
  const quizWordList = new Set(
    [...wordOptions.split(' ')].filter((word: string) => !disallowedWordSet.has(word))
  )
  return [...quizWordList]
}

export function selectQuizWordOptions(correctWord: string, allWords: string[], numOptions: number) {
  const wordOptions = []
  const correctWordPosition = Math.floor(Math.random() * numOptions)
  const randomWordIndexList = sampleSize([...Array(allWords.length).keys()], numOptions - 1)
  let randomWordIndex: number = 0

  for (let i = 0; i < numOptions; i++) {
    if (i === correctWordPosition) {
      wordOptions.push(correctWord)
      continue
    }

    wordOptions.push(allWords[randomWordIndexList[randomWordIndex]])
    randomWordIndex += 1
  }
  return wordOptions
}

export function getWordlist(language: string | null) {
  if (!language) {
    return wordlists.EN
  }

  switch (language.slice(0, 2)) {
    case 'es': {
      return wordlists.ES
    }
    default: {
      return wordlists.EN
    }
  }
}

// TODO(Rossy) Remove after the next alfa testnet reset
export async function getStoredMnemonic(): Promise<string | null> {
  try {
    Logger.debug(TAG, 'Checking keystore for mnemonic')
    let mnemonic = await getKey('mnemonic')
    if (mnemonic) {
      return mnemonic
    }

    Logger.debug(TAG, 'Mnemonic not found in keystore, checking async storage')
    mnemonic = await AsyncStorage.getItem('mnemonic')
    if (mnemonic) {
      await setKey('mnemonic', mnemonic)
      await AsyncStorage.removeItem('mnemonic')
      return mnemonic
    }

    Logger.error(TAG, 'No mnemonic found')
    return null
  } catch (error) {
    Logger.error(TAG, 'Failed to retrieve mnemonic', error)
    return null
  }
}
