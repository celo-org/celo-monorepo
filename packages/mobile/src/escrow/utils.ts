import { ensureLeading0x } from '@celo/base'
import { generateDeterministicInviteCode } from '@celo/utils/lib/account'
import { publicKeyToAddress } from '@celo/utils/lib/address'

export const generateEscrowPaymentIdAndPk = (
  recipientPhoneHash: string,
  recipientPepper: string,
  addressIndex: number = 0
) => {
  const { publicKey, privateKey } = generateDeterministicInviteCode(
    recipientPhoneHash,
    recipientPepper,
    addressIndex
  )
  return { paymentId: publicKeyToAddress(publicKey), privateKey: ensureLeading0x(privateKey) }
}

export const generateUniquePaymentId = (
  existingPaymentIds: string[],
  phoneHash: string,
  pepper: string
) => {
  const paymentIdSet: Set<string> = new Set(existingPaymentIds)

  // Using an upper bound of 100 to be sure this doesn't run forever given
  // the realistic amount of pending escrow txs is far less than this
  for (let i = 0; i < 100; i += 1) {
    const { paymentId } = generateEscrowPaymentIdAndPk(phoneHash, pepper, i)
    if (!paymentIdSet.has(paymentId)) {
      return paymentId
    }
  }
}
