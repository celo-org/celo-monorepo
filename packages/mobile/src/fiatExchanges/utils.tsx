import { SIMPLEX_URI } from 'src/config'
import { CURRENCY_ENUM } from 'src/geth/consts'
import { LocalCurrencyCode } from 'src/localCurrency/consts'
import { navigate } from 'src/navigator/NavigationService'
import { Screens } from 'src/navigator/Screens'
import { navigateToURI } from 'src/utils/linking'

export const openMoonpay = (currencyCode: LocalCurrencyCode, currencyToBuy: CURRENCY_ENUM) => {
  navigate(Screens.MoonPay, {
    localAmount: 0,
    currencyCode,
    currencyToBuy,
  })
}

export const openSimplex = (account: string | null) => {
  navigateToURI(`${SIMPLEX_URI}?address=${account}`)
}
