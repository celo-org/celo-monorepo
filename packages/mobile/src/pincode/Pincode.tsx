/**
 * This is a VIEW. We use it everwhere we need to show PIN pad
 * with an input, e.g. get/ensure/set pincode.
 */
import NumberKeypad from '@celo/react-components/components/NumberKeypad'
import colors from '@celo/react-components/styles/colors.v2'
import fontStyles from '@celo/react-components/styles/fonts.v2'
import React from 'react'
import { ScrollView, StyleSheet, Text, View } from 'react-native'
import PincodeDisplay from 'src/pincode/PincodeDisplay'
import { PIN_LENGTH } from 'src/pincode/utils'

interface Props {
  title?: string
  errorText?: string
  maxLength?: number
  pin: string
  onChangePin: (pin: string) => void
  onCompletePin: (pin: string) => void
}

function Pincode({
  title,
  errorText,
  maxLength = PIN_LENGTH,
  pin,
  onChangePin,
  onCompletePin,
}: Props) {
  function onDigitPress(digit: number) {
    if (pin.length >= maxLength) {
      return
    }

    const newPin = pin + digit
    onChangePin(newPin)
    if (newPin.length === maxLength) {
      // Wait for next frame so we the user can see the last digit
      // displayed before acting on it
      requestAnimationFrame(() => onCompletePin(newPin))
    }
  }

  function onBackspacePress() {
    onChangePin(pin.substr(0, pin.length - 1))
  }

  return (
    <ScrollView style={styles.container} contentContainerStyle={styles.scrollContainer}>
      <View style={styles.spacer} />
      {!errorText && <Text style={styles.title}>{title || ' '}</Text>}
      {!!errorText && <Text style={styles.error}>{errorText}</Text>}
      <View style={styles.pincodeContainer}>
        <PincodeDisplay pin={pin} maxLength={maxLength} />
      </View>
      <View style={styles.spacer} />
      <NumberKeypad onDigitPress={onDigitPress} onBackspacePress={onBackspacePress} />
    </ScrollView>
  )
}

const styles = StyleSheet.create({
  container: {
    marginTop: 40,
  },
  scrollContainer: {
    flexGrow: 1,
  },
  spacer: {
    flex: 1,
  },
  title: {
    ...fontStyles.regular,
    textAlign: 'center',
    marginBottom: 24,
  },
  error: {
    ...fontStyles.regular500,
    color: colors.warning,
    textAlign: 'center',
    marginBottom: 24,
  },
  pincodeContainer: {
    marginBottom: 24,
    paddingHorizontal: '15%',
    alignItems: 'center',
  },
})

export default Pincode
