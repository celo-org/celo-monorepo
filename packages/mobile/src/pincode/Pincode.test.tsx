import * as React from 'react'
import { fireEvent, render } from 'react-native-testing-library'
import { Provider } from 'react-redux'
import Pincode from 'src/pincode/Pincode'
import { createMockStore } from 'test/utils'

describe('Pincode', () => {
  it('renders correctly', () => {
    const { toJSON, getByTestId } = render(
      <Provider store={createMockStore()}>
        <Pincode />
      </Provider>
    )

    // initial - education
    expect(toJSON()).toMatchSnapshot()

    // Press continue
    fireEvent.press(getByTestId('Pincode-Education'))
    expect(toJSON()).toMatchSnapshot()

    fireEvent.press(getByTestId('Pincode-Enter'))
    expect(toJSON()).toMatchSnapshot()

    fireEvent.press(getByTestId('Pincode-ReEnter'))
    expect(toJSON()).toMatchSnapshot()
  })
})
