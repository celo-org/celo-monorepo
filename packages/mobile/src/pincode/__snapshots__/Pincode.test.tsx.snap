// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pincode renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Object {
        "height": 35,
        "paddingRight": 10,
        "paddingTop": 10,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 70,
        "zIndex": 100,
      }
    }
  >
    <Text
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
         
    </Text>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "padding": 20,
        "paddingTop": 0,
      }
    }
  >
    <View>
      <Image
        source={
          Object {
            "testUri": "../../../src/images/backup-icon.png",
          }
        }
        style={
          Array [
            Object {
              "height": 70,
              "width": 80,
            },
            Object {
              "alignSelf": "center",
            },
          ]
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "paddingHorizontal": 10,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2E3338",
                "fontFamily": "Hind-Light",
                "fontSize": 22,
                "paddingBottom": 20,
                "textAlign": "center",
              },
              Object {
                "color": "#2E3338",
                "padding": 25,
                "textAlign": "center",
              },
            ]
          }
          testID="PincodeTitle"
        >
          createPin.title
        </Text>
        <View
          style={
            Object {
              "marginVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#2E3338",
                "fontFamily": "Hind-Regular",
                "fontSize": 14,
                "lineHeight": 18,
              }
            }
          >
            createPin.intro createPin.why
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "marginVertical": 10,
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#2E3338",
                "fontFamily": "Hind-Regular",
                "fontSize": 14,
                "lineHeight": 18,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2E3338",
                    "fontFamily": "Hind-Bold",
                    "fontSize": 14,
                    "lineHeight": 18,
                  },
                ]
              }
            >
              important
               
            </Text>
            createPin.warn
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 3,
            "flexDirection": "row",
          },
          Object {
            "marginVertical": 10,
          },
          Object {
            "marginBottom": 10,
          },
          Object {
            "paddingHorizontal": 20,
            "paddingVertical": 5,
          },
          Object {
            "backgroundColor": "transparent",
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#42D689",
            "borderRadius": 3,
            "borderWidth": 2,
            "flex": 1,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
        testID="Pincode-Education"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Hind-SemiBold",
                  "fontSize": 16,
                },
                Object {
                  "color": "#42D689",
                },
                Object {
                  "paddingLeft": 5,
                  "paddingRight": 5,
                },
              ]
            }
          >
            continue
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Pincode renders correctly 2`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Object {
        "height": 35,
        "paddingRight": 10,
        "paddingTop": 10,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 70,
        "zIndex": 100,
      }
    }
  >
    <Text
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
         
    </Text>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "padding": 20,
        "paddingTop": 0,
      }
    }
  >
    <View>
      <Image
        source={
          Object {
            "testUri": "../../../src/images/backup-icon.png",
          }
        }
        style={
          Array [
            Object {
              "height": 70,
              "width": 80,
            },
            Object {
              "alignSelf": "center",
            },
          ]
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "paddingHorizontal": 10,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2E3338",
                "fontFamily": "Hind-Light",
                "fontSize": 22,
                "paddingBottom": 20,
                "textAlign": "center",
              },
              Object {
                "color": "#2E3338",
                "padding": 25,
                "textAlign": "center",
              },
            ]
          }
        >
          createPin.title
        </Text>
        <TextInput
          allowFontScaling={true}
          autoFocus={true}
          keyboardType="numeric"
          maxLength={6}
          nativeInput={true}
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="createPin.yourPin"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#D1D5D8",
              "borderRadius": 3,
              "borderWidth": 1,
              "fontSize": 24,
              "marginHorizontal": 60,
              "marginVertical": 15,
              "padding": 7,
              "textAlign": "center",
            }
          }
          textContentType="password"
          underlineColorAndroid="transparent"
          validator="integer"
          value=""
        />
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 3,
            "flexDirection": "row",
          },
          Object {
            "marginVertical": 10,
          },
          Object {
            "marginBottom": 10,
          },
          Object {
            "paddingHorizontal": 20,
            "paddingVertical": 5,
          },
          Object {
            "backgroundColor": "#A3EBC6",
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#A3EBC6",
            "borderColor": "#A3EBC6",
            "borderRadius": 3,
            "borderWidth": 2,
            "flex": 1,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
        testID="Pincode-Enter"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Hind-SemiBold",
                  "fontSize": 16,
                },
                Object {
                  "color": "#FFFFFF",
                },
                Object {
                  "paddingLeft": 5,
                  "paddingRight": 5,
                },
              ]
            }
          >
            continue
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Pincode renders correctly 3`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Object {
        "height": 35,
        "paddingRight": 10,
        "paddingTop": 10,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 70,
        "zIndex": 100,
      }
    }
  >
    <Text
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
         
    </Text>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "padding": 20,
        "paddingTop": 0,
      }
    }
  >
    <View>
      <Image
        source={
          Object {
            "testUri": "../../../src/images/backup-icon.png",
          }
        }
        style={
          Array [
            Object {
              "height": 70,
              "width": 80,
            },
            Object {
              "alignSelf": "center",
            },
          ]
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "paddingHorizontal": 10,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2E3338",
                "fontFamily": "Hind-Light",
                "fontSize": 22,
                "paddingBottom": 20,
                "textAlign": "center",
              },
              Object {
                "color": "#2E3338",
                "padding": 25,
                "textAlign": "center",
              },
            ]
          }
        >
          verifyPin.title
        </Text>
        <TextInput
          allowFontScaling={true}
          autoFocus={true}
          keyboardType="numeric"
          maxLength={6}
          nativeInput={true}
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="createPin.yourPin"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#D1D5D8",
              "borderRadius": 3,
              "borderWidth": 1,
              "fontSize": 24,
              "marginHorizontal": 60,
              "marginVertical": 15,
              "padding": 7,
              "textAlign": "center",
            }
          }
          textContentType="password"
          underlineColorAndroid="transparent"
          validator="integer"
          value=""
        />
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 3,
            "flexDirection": "row",
          },
          Object {
            "marginVertical": 10,
          },
          Object {
            "marginBottom": 10,
          },
          Object {
            "paddingHorizontal": 20,
            "paddingVertical": 5,
          },
          Object {
            "backgroundColor": "#42D689",
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#42D689",
            "borderColor": "#42D689",
            "borderRadius": 3,
            "borderWidth": 2,
            "flex": 1,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
        testID="Pincode-ReEnter"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Hind-SemiBold",
                  "fontSize": 16,
                },
                Object {
                  "color": "#FFFFFF",
                },
                Object {
                  "paddingLeft": 5,
                  "paddingRight": 5,
                },
              ]
            }
          >
            verifyPin.finalPin
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Pincode renders correctly 4`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Object {
        "height": 35,
        "paddingRight": 10,
        "paddingTop": 10,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 70,
        "zIndex": 100,
      }
    }
  >
    <Text
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
         
    </Text>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "padding": 20,
        "paddingTop": 0,
      }
    }
  >
    <View>
      <Image
        source={
          Object {
            "testUri": "../../../src/images/backup-icon.png",
          }
        }
        style={
          Array [
            Object {
              "height": 70,
              "width": 80,
            },
            Object {
              "alignSelf": "center",
            },
          ]
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "paddingHorizontal": 10,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2E3338",
                "fontFamily": "Hind-Light",
                "fontSize": 22,
                "paddingBottom": 20,
                "textAlign": "center",
              },
              Object {
                "color": "#2E3338",
                "padding": 25,
                "textAlign": "center",
              },
            ]
          }
        >
          verifyPin.title
        </Text>
        <TextInput
          allowFontScaling={true}
          autoFocus={true}
          keyboardType="numeric"
          maxLength={6}
          nativeInput={true}
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="createPin.yourPin"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#D1D5D8",
              "borderRadius": 3,
              "borderWidth": 1,
              "fontSize": 24,
              "marginHorizontal": 60,
              "marginVertical": 15,
              "padding": 7,
              "textAlign": "center",
            }
          }
          textContentType="password"
          underlineColorAndroid="transparent"
          validator="integer"
          value=""
        />
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 3,
            "flexDirection": "row",
          },
          Object {
            "marginVertical": 10,
          },
          Object {
            "marginBottom": 10,
          },
          Object {
            "paddingHorizontal": 20,
            "paddingVertical": 5,
          },
          Object {
            "backgroundColor": "#42D689",
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#42D689",
            "borderColor": "#42D689",
            "borderRadius": 3,
            "borderWidth": 2,
            "flex": 1,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
        testID="Pincode-ReEnter"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Hind-SemiBold",
                  "fontSize": 16,
                },
                Object {
                  "color": "#FFFFFF",
                },
                Object {
                  "paddingLeft": 5,
                  "paddingRight": 5,
                },
              ]
            }
          >
            verifyPin.finalPin
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
